freetextresponse/freetextresponse.py

```diff
@@ -464,40 +503,64 @@ def _get_submitted_message(self):
         Returns the message to display in the submission-received div
         """
         result = ''
-        if self.count_attempts > 0 and self._word_count_valid():
+        if self._word_count_valid():
             result = self.submitted_message
         return result
 
+    def _get_user_alert(self, ignore_attempts=False):
+        """
+        Returns the message to display in the user_alert(TBD) div
+        """
+        result = ''
+        if not self._word_count_valid():
+            result = self._get_word_count_message(ignore_attempts)
+        return result
+
     @XBlock.json_handler
     def submit(self, data, suffix=''):
         # pylint: disable=unused-argument
         """
         Processes the user's submission
         """
-        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
-            raise StandardError(
-                _(
-                    'User has already exceeded the '
-                    'maximum number of allowed attempts'
-                )
-            )
-        self.student_answer = data['student_answer']
-        if self._word_count_valid():
-            if self.max_attempts == 0:
-                self.count_attempts = 1
-            else:
-                self.count_attempts += 1
-            self._compute_score()
+        if self.max_attempts == 0 or self.count_attempts < self.max_attempts:
+            self.student_answer = data['student_answer']
+            if self._word_count_valid():
+                if self.max_attempts == 0:
+                    self.count_attempts = 1
```

Comment:
> Again, I didn't write this xblock.

I'm aware; I've acknowledged this elsewhere in the thread.

You've now refactored (nearly) this entire submit handler.
When the next person comes along to work on this repo, they're going to `git blame` these lines
and they'll resolve to you; it's your code now, not Azim's.

Look, I'm not blaming you; I'm pointing out that, in lieu of a counter-argument,
this code is wrong. We should fix it while we're editing it anyway.

```
# We don't need to conditionalize this at all based on `max_attempts`
if self._word_count_valid():
    self.count_attempts += 1